Dockerfile

```diff
@@ -0,0 +1,108 @@
+# use builder image to compile ledisdb (without GCO)
+FROM golang:1.9-stretch as builder
+
+ENV LEDISDB_VERSION v0.6
+
+ENV LEVELDB_VERSION 47cb9e2a211e1d7157078ba7bab536beb29e56dc
+ENV ROCKSDB_VERSION 5.8.6
+ENV GOSU_VERSION 1.10
+
+
+
+WORKDIR /build
+
+RUN apt-get update && \
+	apt-get install -y \
+    ca-certificates \
+    wget \
+    gcc-6 \
+    g++-6 \
+    build-essential \
+    libsnappy1v5 \
+    libsnappy-dev \
+    libgflags-dev
+
+# get LedisDB
+RUN wget -O ledisdb-src.tar.gz "https://github.com/siddontang/ledisdb/archive/$LEDISDB_VERSION.tar.gz" && \
+    tar -zxf ledisdb-src.tar.gz && \
+    mkdir -p /go/src/github.com/siddontang/ && \
+    mv ledisdb-$LEDISDB_VERSION /go/src/github.com/siddontang/ledisdb
+
+# build LevelDB
+RUN wget -O leveldb-src.tar.gz "https://github.com/google/leveldb/archive/$LEVELDB_VERSION.tar.gz" && \
+    tar -zxf leveldb-src.tar.gz && \
+    cd leveldb-$LEVELDB_VERSION && \
+    patch -p0 < /go/src/github.com/siddontang/ledisdb/tools/leveldb.patch && \
```

Comment:
do we need this patch now? I think we can remove it if it can't improve too much performance.

Reply:
In your fork you've made a total of 4 changes to LevelDB:
1. you've increased the target (max) file size - which is now exposed as a setting and I've made it configurable in the other branch
2. you've doubled the soft limit on 0-level files (8 -> 16)
3. you've increased the hard cap on 0-level files from 12 to 64
4. you've doubled the grandparent overlap (10 -> 20 times target file size)

Change 1 is no longer required as that value is now exposed in LevelDB and I've made it configurable in Ledis. Changes 2-4 are made using the patch file.

Note: I know next to nothing about LevelDB - all I did was replicating your changes in v0.5.

Reply:
Got it. 

I think we can still keep the patch :-)